Doc 13694 (#4076)
couchbase/docs-server: modules/manage/pages/troubleshoot/common-errors.adoc

[abstract]
{description}

> == File descriptor and core file size limits

Reviewer: ```suggestion
== File Descriptor and Core File Size Limits
```

== File descriptor and core file size limits

If you are having problems starting Couchbase Server on Linux for the first time, there are two common and related causes to consider.
When the [.path]_/etc/init.d/couchbase-server_ script runs, it tries to set the file descriptor limit and core file size limit:

> +

Reviewer: Did deleting all of these `+`'s break any formatting?

Author: No. I've built and checked. The only real change is to make each section an actual section for the sake of navigation and chunking. Honestly the whole doc needs a rethink.

<img width="2693" height="934" alt="image" src="https://github.com/user-attachments/assets/5c072567-88d3-42ad-991d-ce26e4addc28" />

ulimit -n 40960
ulimit -c unlimited
----

This may or may not be allowed depending on your system configuration.
If Couchbase Server is failing to start, you can look through the logs for one or both of these messages:

----
ns_log: logging ns_port_server:0:Port server memcached on node 'ns_1@127.0.0.1' exited with status 71.
Restarting. Messages: failed to set rlimit for open files.
Try running as root or requesting smaller maxconns value.
----

Alternatively, you may additionally see or optionally see:

----
ns_port_server:0:info:message - Port server memcached on node 'ns_1@127.0.0.1' exited with status 71.
Restarting. Messages: failed to ensure core file creation
----

The resolution to these is to edit the /etc/security/limits.conf file and add these entries:

----
couchbase hard nofile 40960
couchbase hard core unlimited
----

> == IP address seems to have changed

Reviewer: ```suggestion
== IP Address Seems To Have Changed
```

Error message:

----
IP address seems to have changed. Unable to listen on 'ns_1@'
----

This message means that Couchbase Server had a problem opening a port with that address.
Couchbase Server listens on all interfaces (0.0.0.0), and chooses which interface to use for its cluster host name (Erlang OTP hostname, not DNS hostname, though the two values can be the same) (ns_1@IP) by determining which interface would lead out to external networks.
This is defined by your routing table.
This error often occurs due to a networking issue, particularly around DHCP and DNS resolution.

> == Cluster version compatibility mismatch

Reviewer: ```suggestion
== Cluster Version Compatibility Mismatch
```

Error message:

----
 This node cannot add another node (ns@xx.xx.xx) because of cluster compatibility mismatch.
 Cluster works in [version xx] mode and only support [version xx].
----

This message means that the cluster is at version XX and only supports adding nodes XX or higher.
Typically, to enable certain features like client-server SSL, or secure XDCR; all the nodes of the cluster to be at a required compatibility level or higher that supports the feature.

> == Difficulties communicating with the cluster. Displaying cached information.

Reviewer: ```suggestion
== Difficulties Communicating With The Cluster. Displaying Cached Information.
```

Error message:

----
Lost connection to server at <IP>:8091. Repeating in XX seconds. Retry now. Repeating failed XHR request...
----

You are seeing the cached UI.
This issue is mostly due to the network connection between your system and the Couchbase Server node.
Verify your connectivity to the Couchbase Server node using [.cmd]`ping` and [.cmd]`telnet`.

> == Transparent huge server pages enabled

Reviewer: ```suggestion
== Transparent Huge Server Pages Enabled
```

      fi
----

> == Blocked view indexing

Reviewer: ```suggestion
== Blocked View Indexing
```

You have defined a view with `a _stats` reduce function.

> == You see constant empty results in your queries to the view

Reviewer: ```suggestion
== Constant Empty Results in Your View Queries
```

       reason: Builtin _stats function requires map values to be numbers
----

> == View time-out errors

Reviewer: ```suggestion
== View Time-Out Errors
```

      ]
      }
----

By default, for queries with `stale=false` (full consistency) the view merging node (node that receives the query request, node 1 in this example) waits up to 60000 milliseconds (1 minute) to receive partial view results from each other node in the cluster.
If it waits for more than 1 minute for results from a remote node, it stops waiting and a timeout error entry is added to the final response.
A `stale=false` request blocks a client, or the view merger node as in this example, until the index is up to date, and these timeouts can happen frequently.

> == Swappiness enabled

Reviewer: ```suggestion
== Swappiness Enabled
```

Make sure that you either have or modify your process that builds your OSs to do this.
This is especially critical for public/private clouds where it is so easy to bring up new instances.
You need to make this part of your build process for a Couchbase node.

== Severe Performance Degradation on Linux Hosts

* Sudden, severe throughput reduction and/or latency spikes on specific Linux hosts.
* Often correlated with Intel CPUs and Linux kernels that enable split lock detection/mitigation.

If the CPU supports split lock detection, a misaligned atomic memory access can trigger Linux split lock mitigation.
The mitigation can artificially slow execution, causing a pronounced performance drop for the affected workload.

> For mitigating information, see the article on xref:install:install-splitlock-mitigation.adoc[]

Reviewer: ```suggestion
For more information about how to resolve issues with split lock mitigation, see xref:install:install-splitlock-mitigation.adoc[].
```

Author: Thanks. Fixed.

Make sure that you either have or modify your process that builds your OSs to do this.
This is especially critical for public/private clouds where it is so easy to bring up new instances.
You need to make this part of your build process for a Couchbase node.

== Severe Performance Degradation on Linux Hosts

* Sudden, severe throughput reduction and/or latency spikes on specific Linux hosts.
* Often correlated with Intel CPUs and Linux kernels that enable split lock detection/mitigation.

=== Cause

> If the CPU supports split lock detection, a misaligned atomic memory access ("split lock") can trigger Linux split lock mitigation.

Reviewer: Same critique about using the term before defining it

The mitigation can artificially slow execution, causing a pronounced performance drop for the affected workload.

=== Resolution

Disable split lock mitigation on the affected Couchbase Server host.

. Permanent (boot-time) fix
+
Add `split_lock_detect=off` to the kernel command line and reboot the host.
+
[source,bash]
----
# Example kernel cmdline parameter
split_lock_detect=off
----

. Immediate (runtime) mitigation (kernel-dependent)
+
If present on your kernel, you can disable mitigation via `/proc`:
+
[source,bash]
----
echo 0 > /proc/sys/kernel/split_lock_mitigate

> ----

Reviewer: Do we want to repeat this or just link to the page with details? No need for a partial, just a link.

Author: Fair. I will leave some details on the symptoms and link to the fix.
